### ISSUE TITLE
Document method to Update all modified properties

### ISSUE THREAD

**Issue by govindrai** [Automattic/mongoose#7160]
I am wondering if there is a method to **_update_** all modified properties of a document (`Document.prototype.modifiedPaths()` beautifully describes changed properties). I know `Document.prototype.save` does this but it runs a bunch of middleware I don't need, I just want to update, but don't want to have to explicitly set properties to update which is the case with `Document.prototype.updateOne`. Thanks.

**Comment by vkarpov15** [comment]
So you just want to basically run save() but skip all middleware?

**Comment by vkarpov15** [comment]
That's an interesting idea. What middleware do you want to skip? Mongoose's built-in middleware or your own middleware?

**Comment by govindrai** [comment]
@vkarpov15 finally got around to this. I wouldn't want to skip any middleware (i.e. I think `document.prototype.updateOne's` pre/post hooks are very useful). I only stated "skip middleware" in my question to describe my usecase (i.e. I want `document.prototype.updateOne` to behave like `document.prototype.save` which automatically writes modified properties, but doesn't come with the middleware baggage that comes with `document.prototype.save`). 

I know I could write some conditionals that return out of the middlewares and just do a save, but it pollutes the codebase and doesn't feel right either. I think it would be super useful for mongoose users if `document.prototype.updateOne` could issue an update with all modified paths if an `update` object argument is not supplied (in fact, I assumed that mongoose functioned as such by default but found otherwise). 

I think this change is worth looking into--I would think the usual usecase in most applications on an update would be: one or more functions run to update one or more properties on a document and instead of needing to loop through `document.prototype.modifiedPaths` which also unfortunately lists top level and intermediary paths (https://github.com/Automattic/mongoose/issues/7373), you could just issue the updateOne command an it takes care of the update, just like `document.prototype.save`.

**Comment by vkarpov15** [comment]
It sounds like you're just trying to implement your own `save()`, "issue an update with all modified paths" is exactly what [`save()`](https://mongoosejs.com/docs/api.html#document_Document-save) does. What middleware baggage comes with `save()` that doesn't come with `updateOne()`?

**Comment by govindrai** [comment]
@vkarpov15 Sorry I never got around to responding. I'm no longer working on this project anymore but since you kept this open for so long, it deserves a response.

Here's what I think I was trying to state:
I expected Document.prototype.update to be able to intelligently update the document based on its modified paths, (like save does).

Desired Behavior:
```
doc.name = 'foo';
doc.update();
// Mongoose would send an `updateOne({ _id: doc._id }, { $set: { name: 'foo' } })` to MongoDB.
```

However that does nothing at all, as the update requires a query object to be passed. I don't want to have to explicitly set properties to update which is the case with Document.prototype.updateOne. I know I could use save, but I don't want to run my save's middlewares, I want to run my updateOne middlewares instead.

**Comment by AbdelrahmanHafez** [comment]
@govindrai 

Not that I recommend going on production with that, but I believe this achieves the desired results:
```js
  const User = mongoose.model('User', userSchema);
  const user = await User.create({ name: 'Hafez', age: 25 });

  user.age = 26;

  const changes = user.$__delta()[1]; // { $set: { age: 26 } }
  await user.update(changes);
```

We're currently considering a feature of an option to disable middlewares on specific operations: https://github.com/Automattic/mongoose/issues/8768

**Comment by govindrai** [comment]
@AbdelrahmanHafez that's definitely a solution, didn't know you could calculate a delta like that! but code becomes ugly real quick :) I definitely like the option of being able to disable middleware hooks but I wouldn't want to disable hooks in this case. I would want all my update hooks to be ran, I just don't want to have to supply the deltas myself.

**Comment by vkarpov15** [comment]
We should really add a function that gets the update to the public API. I wouldn't recommend using `$__delta()` because that's not part of the public API.

**Comment by vkarpov15** [comment]
Closed by #9096
